test/unit/index.js

```diff
@@ -73,15 +73,19 @@ describe('minimal-cipher', function() {
           () => Math.floor(Math.random() * 255));
       }
 
-      async function encryptStream({data}) {
+      async function encryptStream({data, queSize = 5, chunkSize = 5}) {
         const stream = new ReadableStream({
           pull(controller) {
-            controller.enqueue(data);
+            // break the unit8Array into chunks using queSize
+            for(let i = 0; i < data.length; i += queSize) {
+              const chunk = data.slice(i, i + queSize);
```

Comment:
It appears to me that the `chunkSize` is a function of `data.length` and `queSize`.  Does `chunkSize` need to be passed in here, or should it be computed?

I believe it's the case that the last chunk may not be the same size if `data.length` is not evenly divisible by `queSize`, does the present any issues since `chunkSize` appears to be a static variable.

Also, this might have some application here: https://lodash.com/docs/4.17.15#chunk

Reply:
Hey good Pr comments over all. I believe there might be a small misunderstanding here. chunkSize is used by `createEncryptStream` to decide on how the encrypted data is chunked in terms of bytes:

```js
* @param {number} [options.chunkSize=1048576] - The size, in bytes,
*   of the chunks to break the incoming data into.
```

`queueSize` on the other hand is used to take unencrypted data in the form of a Uint8Array, then to take that data in enqueue it several steps. queueSize in other words simulates the way a browser might enqueue a 2 megabyte file as a set of 10kb chunks. This is independent of the `chunkSize` set in `createEncryptStream`.

A big part of what is being tested here is that if a device passes an unencrypted stream that say contains 20 10k chunks, and we create an encrypted stream with a chunkSize of say 25kb then we should expect the encrypted stream to have a different number of enqueued chunks which should still decrypt to the same UInt8Array passed in originally. There are 2 streams here, an unencrypted data stream representing a file or some other user input, and then an encryptedStream which is what we actually push to the server. In other words chunkSize and queueSize are different because they are part of the tests being performed in this suite.

Reply:
OK, after your explanation and studying the implementation, I think this helper is doing to much which is what leads to my confusion about what the parameters are about.  I think it would be more clear if this were separated into two functions.

`createReadableStream` which encapsulates this code and returns a stream: https://github.com/digitalbazaar/minimal-cipher/blob/5583046fb336464f46cb5ce7b5dd14de660d52ad/test/unit/index.js#L77-L86

The stream is then passed into `encryptStream`.  

This would serve to separate the `queueSize` and `chunkSize` parameters into their separate helpers.

How does that sound?

Reply:
I like this idea and I added comments and a new helper method. I did not want to rewrite every test to pass in a ReadableStream, but the resulting code I think makes the test library more understandable: https://github.com/digitalbazaar/minimal-cipher/pull/29/commits/ac8b818c13aad3abcf70859676bb8ba8433d26bc

Reply:
Thanks.